core/src/main/java/org/apache/iceberg/avro/ValueWriters.java

```diff
@@ -492,4 +561,153 @@ protected Object get(IndexedRecord struct, int pos) {
       return struct.get(pos);
     }
   }
+
+  private abstract static class FloatingPointWriter<T extends Comparable<T>>
+      extends ComparableWriter<T> {
```

Comment:
I don't think this needs to extend `ComparableWriter` since it calls `compareTo` directly.

Reply:
I think this was to mainly save an argument for constructor, but I can pass `null` and add a comment.